packages/hash/api/src/model/aggregation.model.ts

```diff
@@ -158,31 +158,47 @@ class __Aggregation {
 
           if (typeof item !== "string") return null;
 
+          if (
+            [
+              "CONTAINS",
+              "DOES_NOT_CONTAIN",
+              "IS",
+              "IS_NOT",
+              "ENDS_WITH",
+              "STARTS_WITH",
+            ].includes(filterItem.operator) &&
+            !filterItem.value
+          ) {
+            throw new Error(
+              `You must provide 'value' when using operator '${filterItem.operator}'`,
+            );
+          }
+
           switch (filterItem.operator) {
             case "CONTAINS":
               return item
                 .toLowerCase()
-                .includes(filterItem.value.toLowerCase());
+                .includes(filterItem.value!.toLowerCase());
```

Comment:
Ideally we could do this in a more type-safe way (moving all of the operators that require a value into a branch that checks that value exists, so that there isn't any need for these assertions), but that can wait for a future PR.

Reply:
Yes, this is suboptimal. We will remove this filtering logic before long in favour of filtering in the db, so hopefully we don't have to revisit.